zebra/zebra_pbr.c

```diff
@@ -156,118 +156,43 @@ void zebra_pbr_rules_free(void *arg)
 
 uint32_t zebra_pbr_rules_hash_key(const void *arg)
 {
-	const struct zebra_pbr_rule *rule;
+	const struct zebra_pbr_rule *rule = arg;
 	uint32_t key;
 
-	rule = arg;
-	key = jhash_3words(rule->rule.seq, rule->rule.priority,
-			   rule->rule.action.table,
-			   prefix_hash_key(&rule->rule.filter.src_ip));
-
-	key = jhash_3words(rule->rule.filter.fwmark, rule->vrf_id,
-			   rule->rule.filter.ip_proto, key);
+	key = jhash_2words(rule->rule.unique, rule->sock, 0);
```

Comment:
Having some random init value for the 'key' would be nice - it's a good practice to follow?

Reply:
Is that the same thing as a "seeded" hash? It looks like that is a defense against DDOS attacks so I'm guessing in this case an attacker would craft a set of PBR rules to slow down lookups?

In general it could be a good thing. Maybe better implemented in the hash library so all callers benefit?

Reply:
My point is just about having the various examples in the codebase stay aligned as much as possible. In general, we try to encourage some kind of initialization for the key, so it'd be good to do that here also.

Reply:
Please help me understand - my eyeball estimate is that around a third of the jhash_ algorithms in FRR start with some arbitrary fixed seed, and the rest use 0. What does it accomplish? What's the basis for choosing a particular 32-bit seed?

I'm not finding either general information (via goog) or any comments in frr/docs/developer.